cirq/experiments/n_qubit_tomography.py

```diff
@@ -0,0 +1,194 @@
+
+        Returns:
+            TomographyResult with density matrix corresponding to solution of
+            this system.
+        """
+        # normalize the input.
+        probs = counts / np.sum(counts, axis=1)[:, np.newaxis]
+        # use least squares to get solution.
+        c, _, _, _ = np.linalg.lstsq(self.mat, np.asarray(probs).flat, rcond=-1)
+        rho = c.reshape((2**self.num_qubits, 2**self.num_qubits))
+        return TomographyResult(rho)
+
+
+def state_tomography(
+        sampler: cirq.Sampler,
+        qubits: List[cirq.Qid],
+        circuit: cirq.Circuit,
+        repetitions: int = 1000,
+        prerotations: List[Tuple[float, float]] = None,
+) -> 'TomographyResult':
+    """This performs n qubit tomography on a cirq circuit
+
+    Follows https://web.physics.ucsb.edu/~martinisgroup/theses/Neeley2010b.pdf
+    A.1. State Tomography.
+    This is a high level interface for StateTomographyExperiment.
+    Args:
+        circuit: circuit to do the tomography on.
+        qubits: qubits to do the tomography on.
+        sampler: sampler to collect the data from.
+        repetitions: number of times to sample each rotations
+        prerotations: Tuples of (phase_exponent, exponent) parameters for gates
+            to apply to the qubits before measurement. The actual rotation
+            applied will be
+                cirq.PhasedXPowGate(
+                    phase_exponent=phase_exponent, exponent=exponent)
+            If none, we use [(0, 0), (0, 0.5), (0.5, 0.5)], which corresponds to
+            rotation gates [I, X**0.5, Y**0.5]
+    Returns:
+        Tomography result which contains the density matrix of the qubits
+        determined by tomography.
+    """
+    if prerotations == None:
+        prerotations = [(0, 0), (0, 0.5), (0.5, 0.5)]
+
+    exp = StateTomographyExperiment(qubits, prerotations)
+    rot_moment, rot_sweep = (exp.rot_moment, exp.rot_sweep)
+    probs = get_state_tomography_data(sampler,
+                                      qubits,
+                                      circuit,
+                                      rot_moment,
+                                      rot_sweep,
+                                      repetitions=repetitions)
+    return exp.fit_density_matrix(probs)
+
+
+def get_state_tomography_data(sampler: cirq.Sampler,
+                              qubits: List[cirq.Qid],
+                              circuit: cirq.Circuit,
+                              rot_moment: cirq.Circuit,
+                              rot_sweep: cirq.Sweep,
+                              repetitions: int = 1000) -> np.ndarray:
+    """Gets the data for each rotation string added to the circuit.
+
+    For each sequence in prerotation_sequences gets the probability of all
+    2**n bit strings.  Resulting matrix will have dimensions
+    (len(rot_sweep)**n, 2**n).
+    This is a default way to get data that can be replaced by the user if they
+    have a more advanced protocol in mind.
```

Comment:
If this function were a method of `StateTomographyExperiment` it would be easier for users to override.

Reply:
I think it makes more sense for us to have it separate.  The class is designed to hold the matrix and fit the data but not do the actual collection.